crates/cli/cli/src/commands/query/packet/pending.rs

```diff
@@ -0,0 +1,175 @@
+        value_name = "CHANNEL_ID",
+        help_heading = "REQUIRED"
+    )]
+    channel_id: ChannelId,
+}
+
+/// A structure to display pending packet commitment sequence IDs
+/// at both ends of a channel.
+#[derive(Debug, Serialize)]
+struct Summary<P> {
+    /// Source chain
+    src_chain: ChainId,
+
+    /// Destination chain
+    dst_chain: ChainId,
+
+    /// The packets sent on the source chain as identified by the command.
+    src: P,
+
+    /// The packets sent on the counterparty chain.
+    dst: P,
+}
+
+impl Summary<PendingPackets> {
+    fn collate(self) -> Summary<CollatedPendingPackets> {
+        Summary {
+            src_chain: self.src_chain,
+            dst_chain: self.dst_chain,
+
+            src: CollatedPendingPackets::new(self.src),
+            dst: CollatedPendingPackets::new(self.dst),
+        }
+    }
+}
+
+impl fmt::Display for Summary<CollatedPendingPackets> {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        writeln!(f, "Summary of pending packets:")?;
+
+        writeln!(f, "Packets pending on source chain ({}):", self.src_chain)?;
+        writeln!(f, "  Unreceived packets:")?;
+        for seq in &self.src.unreceived_packets {
+            writeln!(f, "    {}", seq)?;
+        }
+        writeln!(f, "  Unreceived acks:")?;
+        for seq in &self.src.unreceived_acks {
+            writeln!(f, "    {}", seq)?;
+        }
+
+        writeln!(
+            f,
+            "Packets pending on destination chain ({}):",
+            self.dst_chain
+        )?;
+        writeln!(f, "  Unreceived packets:")?;
+        for seq in &self.dst.unreceived_packets {
+            writeln!(f, "    {}", seq)?;
+        }
+        writeln!(f, "  Unreceived acks:")?;
+        for seq in &self.dst.unreceived_acks {
+            writeln!(f, "    {}", seq)?;
+        }
+
+        Ok(())
+    }
+}
+
+impl QueryPendingPackets {
+    async fn execute(&self, builder: CosmosBuilder) -> Result<Summary<PendingPackets>> {
+        let chain_id = self.chain_id.clone();
+        let port_id = self.port_id.clone();
+        let channel_id = self.channel_id.clone();
+        let chain = builder.build_chain(&self.chain_id).await?;
+
+        let chan_conn_cli = chain
+            .with_blocking_chain_handle(move |handle| {
+                let chan_conn_cli = channel_connection_client(&handle, &port_id, &channel_id)
+                    .map_err(|e| {
+                        BaseError::generic(eyre!("failed channel_connection_client: {}", e))
+                    })?;
+                Ok(chan_conn_cli)
+            })
+            .await
+            .map_err(|e| {
+                BaseError::generic(eyre!("failed pending_packet_summary on source: {}", e))
+            })?;
+
+        let counterparty_chain_id = chan_conn_cli.client.client_state.chain_id();
+        let counterparty_chain = builder.build_chain(&counterparty_chain_id.clone()).await?;
+
+        let src_summary = pending_packet_summary(
```

Comment:
We have implemented `query_unreceived_packet_sequences` in Hermes SDK. However we have not yet implemented the query for unreceived acks. So I supposed we have to use the v1 `pending_packet_summary` here for now.

Perhaps @ljoss17 can help implement the unreceived acks query later so that we can migrate to the new code?

Reply:
opened #160

Reply:
I have implement `query_unreceived_acks_sequences` here: https://github.com/informalsystems/hermes-sdk/pull/161

What's left is to define something similar to:

https://github.com/informalsystems/hermes/blob/master/crates/relayer/src/chain/counterparty.rs?plain=1#L581